cpp_tests/feature/one_hot_encoder_test.cc

```diff
@@ -14,79 +14,104 @@
  * limitations under the License.
  */
 
-#include "fstream"
-#include <cstdlib>
-#include <string>
-#include <sys/stat.h>
-
 #include "clink/cpp_tests/test_util.h"
 #include "clink/feature/one_hot_encoder.h"
-#include "nlohmann/json.hpp"
 #include "gtest/gtest.h"
 
-namespace tfrt {
+namespace clink {
+
 namespace {
 
 TEST(OneHotEncoderTest, Param) {
-  clink::OneHotEncoderModel model;
-  model.setDropLast(false);
-  EXPECT_EQ(model.getDropLast(), false);
-  model.setDropLast(true);
-  EXPECT_EQ(model.getDropLast(), true);
+  RCReference<OneHotEncoderModel> model =
+      tfrt::TakeRef(test::test_host_context->Construct<OneHotEncoderModel>(
+          test::test_host_context.get()));
+  model->setDropLast(false);
+  EXPECT_FALSE(model->getDropLast());
+  model->setDropLast(true);
+  EXPECT_TRUE(model->getDropLast());
 }
 
 TEST(OneHotEncoderTest, Transform) {
-  clink::OneHotEncoderModelDataProto model_data;
+  OneHotEncoderModelDataProto model_data;
   model_data.add_featuresizes(2);
   model_data.add_featuresizes(3);
   std::string model_data_str;
   model_data.SerializeToString(&model_data_str);
 
-  clink::OneHotEncoderModel model;
-  model.setDropLast(false);
-  llvm::Error err = model.setModelData(model_data_str);
-  EXPECT_EQ(!err, true);
+  RCReference<OneHotEncoderModel> model =
+      tfrt::TakeRef(test::test_host_context->Construct<OneHotEncoderModel>(
+          test::test_host_context.get()));
+  model->setDropLast(false);
+  llvm::Error err = model->setModelData(std::move(model_data_str));
+  EXPECT_FALSE(err);
 
-  auto vector = model.transform(1, 0);
-  EXPECT_EQ(!vector.takeError(), true);
+  auto vector = model->transform(1, 0);
+  EXPECT_TRUE(vector.IsConcrete());
   EXPECT_EQ(vector->get(1).get(), 1.0);
 
-  auto invalid_value_vector = model.transform(1, 5);
-  EXPECT_EQ(!invalid_value_vector.takeError(), false);
+  auto invalid_value_vector = model->transform(1, 5);
+  EXPECT_FALSE(invalid_value_vector.IsConcrete());
```

Comment:
Is the returned `invalid_value_vector` expected to be immediately available with error? If so, would it be better to use `EXPECT_TRUE(invalid_value_vector.IsError())`?

Same for `invalid_index_vector` and other usages of `IsConcrete(...)`.

Reply:
Yes. For one hot encoder these methods are supposed to return immediately. I'll make the changes.